### PR TITLE
Enable tests and coverage reports

### PR DESCRIPTION
**Pull request by przemyslawzalewski**
Tests can now be written and run by jest. Some issues are still present as the built-in babel-jest transpilation causes instrumentation to be less effective compared to pure JS one, especially when it comes to the default parameter coverage.

This PR opens a way to cover the already existing code with tests by providing some example test files. Most of the tests are skipped for now and are subject to continuous improvements.

Coverage generation is now turned on, so coverage tools can now be enabled so the progress and degradations can be listed right within a PR. Local reports are generated and available at `./.coverage/lcov-report/index.html`:

chrome_2019-12-05_10-38-08

chrome_2019-12-05_10-38-16

chrome_2019-12-05_10-38-21

### CODE REVIEW

File: ignore.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const ignoredFiles = [
   ".all-contributorsrc",
+  ".babelrc.js",
```
Comment (marekrozmus) [inline ignore.js]: consider migrate to babel.config.js

File: array/are.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+describe("are", () => {
+  it("returns true if the given arguments are all arrays", () => {
+    expect(are([2, 3])).toBe(true);
+    expect(are([1, 2, 3], [])).toBe(true);
+    expect(are([1, 2, 3], 8, [1, 3], "test")).toBe(false);
+  });
+
+  it("returns true for empty list of arguments", () => {
+    expect(are()).toBe(true);
+  });
```
Comment (adamklepacz) [inline array/are.test.ts]: why it returns true for no arguments?

I would expect to return true only when arrays are passed. 
False for array mixed with other data types, no arguments, null, undefned, Number, String as argument.

File: ignore.js (hunk 1 above)
Comment (przemyslawzalewski) [reply ignore.js]: I have investigated that and we gain nothing vs the `.babelrc.js` format so I'm gonna skip that for now.

File: index.test.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
+/* eslint-env jest */
 // @ts-check
-import "./index.js";
-import { add } from "./math/index.js";
+import std from "./index.js";
 
-add(5, 3);
+test("the module be defined", () => {
```
Comment (adamklepacz) [inline index.test.js]: `the module is defined`

File: index.test.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
-import "./index";
-import { add } from "./math";
+/* eslint-env jest */
+import std from "./index";
 
-add(5, 3);
+test("the module should be defined", () => {
```
Comment (adamklepacz) [inline index.test.ts]: `module is defined`

File: array/are.test.ts (hunk 2 above)
Comment (przemyslawzalewski) [reply array/are.test.ts]: Just to be consistent with `[].every(Array.isArray)` which is what the `are` is meant to do. This is just a plural of `array/is` function.

File: index.test.js (hunk 3 above)
Comment (przemyslawzalewski) [reply index.test.js]: This test is now skipped and will be removed in the future. Thanks for pointing this out.

**Pull request merged**